internal/engine/upper_test.go

```diff
@@ -878,19 +937,42 @@ func newTestFixture(t *testing.T) *testFixture {
 	}
 
 	return &testFixture{
-		f,
-		ctx,
-		cancel,
-		upper,
-		b,
-		watcher,
-		&timerMaker,
-		docker,
-		hud,
-		podEvents,
+		TempDirFixture: f,
+		ctx:            ctx,
+		cancel:         cancel,
+		upper:          upper,
+		b:              b,
+		fsWatcher:      watcher,
+		timerMaker:     &timerMaker,
+		docker:         docker,
+		hud:            hud,
+		podEvents:      podEvents,
+		log:            log,
 	}
 }
 
+func (f *testFixture) Start(manifests []model.Manifest, watchMounts bool) {
```

Comment:
I feel like this Start/Stop is a bit cleaner than the way we've been writing upper tests:
1. no `go` in the tests themselves
2. canceling the context instead of feeding sentinels through watch errors
3. removes a bit of redundancy

I haven't yet converted existing tests to this.

Reply:
ya, i like this pattern!